index.d.ts

```diff
@@ -58,15 +62,11 @@ export function useStore<SomeStore extends Store>(
  *     `keys` attribute controls which store value properties will be returned and listened to.
  * @returns Store value.
  */
-export function useStore<
-  SomeStore extends Store,
-  Key extends AllKeys<StoreValue<SomeStore>>
->(
+export function useStore<SomeStore extends Store>(
   store: SomeStore,
-  options?: UseStoreOptions<SomeStore, Key>
-): SomeStore extends MapStore
-  ? Pick<StoreValue<SomeStore>, Key>
-  : StoreValue<SomeStore>
```

Comment:
As I understand (correct me if I'm wrong) that was an error in types and, overall, an incorrect narrowing of the union. The fact that you only listen to certain keys does not mean you will get a narrowed entity. At least that's not how the runtime of `listenKeys` is implemented: since we always listen to `undefined` [here](https://github.com/nanostores/nanostores/blob/main/listen-keys/index.js#L2), it means that when you call `.set` instead of `.setKey` you will get an update here. So narrowing is not correct.

Also, I believe, `useSyncExternalStore` calls `get` from time to time, so you can get a current state here as well, that completely ignores keys you subscribed to.